compiler/src/iree/compiler/Codegen/LLVMGPU/test/reduction_pipeline_cuda.mlir

```diff
@@ -195,7 +195,6 @@ hal.executable.variant @cuda target(<"cuda", "cuda-nvptx-fb">) {
 //         CHECK:    arith.maxnumf
 //         CHECK:    gpu.shuffle  xor
 //         CHECK:    arith.maxnumf
-//         CHECK:    arith.maxnumf
 //         CHECK:    vector.broadcast %{{.*}} : f32 to vector<4xf32>
```

Comment:
What happened here?

Reply:
I am not sure, that manumf is not present in IR, overall a lot of reduction codegen seems changed. I wonder if its something with valuebound interface like this commit
https://github.com/llvm/llvm-project/pull/122804
I will flag this on dicord.

Reply:
Actually I think whats happening here can be explained by this commit
https://github.com/llvm/llvm-project/pull/118952
I believe that we were using -INF in softmax decompostion when we wanted NAN, with NAN, I think there is some simplification as you would simply pick the other argument hence needing one less maxnumf after folding..

Reply:
lets check what happens to correctness with this... Might need to revert it. I missed it, but I dont think using NAN make sense. Should be -INF.

Reply:
The CI is not finding any errors, I believe there are softmax dispactches checked into the regression tests so I think this works. I think there is some discussion on this here
https://github.com/llvm/llvm-project/pull/114595
NaN is safe with maxnumf but should we have used maximumf and kept using -INF would be something to decide. Also should manumf fold the same for -INF is also something to think over.